frontend/src/components/pages/acls/Acl.List.tsx

```diff
@@ -441,6 +403,72 @@ const UsersTab = observer(() => {
   );
 });
 
+const UserActions = ({ user }: { user: UsersEntry }) => {
+  const [isChangePasswordModalOpen, setIsChangePasswordModalOpen] = useState(false);
+  const [isChangeRolesModalOpen, setIsChangeRolesModalOpen] = useState(false);
+
+  const onConfirmDelete = async () => {
+    await api.deleteServiceAccount(user.name);
+
+    // Remove user from all its roles
+    const promises = [];
+    for (const [roleName, members] of rolesApi.roleMembers) {
+      if (members.any((m) => m.name === user.name)) {
+        // is this user part of this role?
+        // then remove it
+        promises.push(rolesApi.updateRoleMembership(roleName, [], [user.name]));
+      }
+    }
+
+    await Promise.allSettled(promises);
+    await rolesApi.refreshRoleMembers();
+    await api.refreshServiceAccounts();
+  };
+
+  return (
+    <>
+      <ChangePasswordModal
+        userName={user.name}
+        isOpen={isChangePasswordModalOpen}
+        setIsOpen={setIsChangePasswordModalOpen}
+      />
+      <ChangeRolesModal userName={user.name} isOpen={isChangeRolesModalOpen} setIsOpen={setIsChangeRolesModalOpen} />
+
+      <Menu>
+        <MenuButton as={Button} variant="ghost" className="deleteButton" style={{ height: 'auto' }}>
+          <Icon as={BsThreeDots} />
+        </MenuButton>
+        <MenuList>
+          <MenuItem
+            onClick={(e) => {
+              e.stopPropagation();
+              setIsChangePasswordModalOpen(true);
+            }}
+          >
+            Change password
+          </MenuItem>
+          {Features.rolesApi && (
+            <MenuItem
+              onClick={(e) => {
+                e.stopPropagation();
+                setIsChangeRolesModalOpen(true);
+              }}
+            >
+              Change roles
+            </MenuItem>
+          )}
+          <DeleteUserConfirmModal
+            onConfirm={onConfirmDelete}
+            buttonEl={<MenuItem type="button">Delete</MenuItem>}
+            userName={user.name}
+          />
```

Comment:
Should we move the modal away from the menu list? Just below the menu, as we are using a react fragment

Reply:
Oh I remember, then let's leave it as-is, thanks for the reminder.